### PR TITLE
mock-tasks-task-filtering-methods-charles

### PR DESCRIPTION
**Pull request by Yuxuan-Wu**
## Feature/Change Description
Mock tasks created

## Developer Contributions
@Yuxuan-Wu 

## Dependencies
#48 #65

### CODE REVIEW

File: src/js/library.js
Comment (davidtso1219) [inline src/js/library.js:1]: Let's remove this for now. This is not how we should do it. We probably shouldn't pass the array as an argument. Instead, a filterTasks function in index.js of the home page should manage this.

**Review by davidtso1219 — CHANGES_REQUESTED** [review]

File: src/js/library.js
Comment (Yuxuan-Wu) [reply src/js/library.js:1]: The method has been removed and replaced by filtering methods in /utils/task-filtering.js

File: src/utils/task-filtering.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * @param {Array} tasks - The array of tasks to be filtered.
+ * @param {string} status - The status to filter by.
+ * @returns {Array} The filtered array of tasks.
+ */
+function filterTasksByStatus(tasks, status) {
+  return tasks.filter(task => task.status === status);
+}
+
+/**
+ * Filter an array of tasks based on their creation time.
```
Comment (sbogh) [inline src/utils/task-filtering.js:24]: I think personally the only filter we need is for oldest to newest and newest to oldest, I would like others input but I wouldn't think its necessary to filter by a specific date (like before 12/5 or after). That would also introduce a good bit of overhead to do error handling for whatever the user inputs @davidtso1219 @mottur let me know your thoughts

File: src/utils/task-filtering.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/**
```
Comment (sbogh) [inline src/utils/task-filtering.js:1]: great job commenting your functions! this is perfect and gives people easy understanding of what each does.

File: test/task-filtering.test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import { filterTasksByType, filterTasksByStatus, filterTasksByCreationTime } from '../src/utils/task-filtering';
+import mockTasks from '../src/mock/mock-tasks.json';
+
+describe('Task Filtering Functions', () => {
```
Comment (sbogh) [inline test/task-filtering.test.js:4]: I would recommend adding comments for each test case here explaining what it is testing for, what type of test it us (error handling, functionality test, etc), and what the expected outcome is

Comment (sbogh) [reply test/task-filtering.test.js:4]: You kind of already have the information in the code itself but it would be better to specifically spell it out

**Review by sbogh — CHANGES_REQUESTED** [review]
Good starting point, just some small commenting behaviors that need to be added. Well done!

@davidtso1219 @mottur please take a look at the comment I referenced you in I would like your feedback.

File: src/utils/task-filtering.js (hunk 1 above)
Comment (mottur) [reply src/utils/task-filtering.js:24]: i agree, newest to oldest and vice versa is probably all that's needed.

File: test/task-filtering.test.js (hunk 3 above)
Comment (mottur) [reply test/task-filtering.test.js:4]: As Shayan said, please add at least one comment per test case discussing what kind of test it is.

File: src/utils/task-filtering.js (hunk 1 above)
Comment (davidtso1219) [reply src/utils/task-filtering.js:24]: yea this function is unlikely to be needed

**Review by davidtso1219 — CHANGES_REQUESTED** [review]

**Review by davidtso1219 — APPROVED** [review]

**Review by sbogh — APPROVED** [review]

**Pull request merged**